src/thor/unidirectional_astar.cc

```diff
@@ -355,34 +518,55 @@ TimeDepForward::GetBestPath(valhalla::Location& origin,
   }
   return {}; // Should never get here
 }
 
+template std::vector<std::vector<PathInfo>>
+UnidirectionalAStar<ExpansionType::forward>::GetBestPath(valhalla::Location& origin,
+                                                         valhalla::Location& destination,
+                                                         GraphReader& graphreader,
+                                                         const sif::mode_costing_t& mode_costing,
+                                                         const TravelMode mode,
+                                                         const Options& /*options*/);
+template std::vector<std::vector<PathInfo>>
+UnidirectionalAStar<ExpansionType::reverse>::GetBestPath(valhalla::Location& origin,
+                                                         valhalla::Location& destination,
+                                                         GraphReader& graphreader,
+                                                         const sif::mode_costing_t& mode_costing,
+                                                         const TravelMode mode,
+                                                         const Options& /*options*/);
+// Set the mode and costing
 // Initialize prior to finding best path
-void TimeDepForward::Init(const midgard::PointLL& origll, const midgard::PointLL& destll) {
-  LOG_TRACE("Orig LL = " + std::to_string(origll.lat()) + "," + std::to_string(origll.lng()));
-  LOG_TRACE("Dest LL = " + std::to_string(destll.lat()) + "," + std::to_string(destll.lng()));
-
-  // Set the destination and cost factor in the A* heuristic
-  astarheuristic_.Init(destll, costing_->AStarCostFactor());
-
-  // Get the initial cost based on A* heuristic from origin
-  float mincost = astarheuristic_.Get(origll);
-
-  // Reserve size for edge labels - do this here rather than in constructor so
-  // to limit how much extra memory is used for persistent objects.
-  // TODO - reserve based on estimate based on distance and route type.
+template <const ExpansionType expansion_direction, const bool FORWARD>
+void UnidirectionalAStar<expansion_direction, FORWARD>::Init(const midgard::PointLL& origll,
+                                                             const midgard::PointLL& destll) {
+
+  float mincost = 0;
+  if (FORWARD) {
+    astarheuristic_.Init(destll, costing_->AStarCostFactor());
+    mincost = astarheuristic_.Get(origll);
+  } else {
+    astarheuristic_.Init(origll, costing_->AStarCostFactor());
+    mincost = astarheuristic_.Get(destll);
+  }
   edgelabels_.reserve(std::min(max_reserved_labels_count_, kInitialEdgeLabelCount));
 
   // Construct adjacency list, clear edge status.
   // Set bucket size and cost range based on DynamicCost.
   uint32_t bucketsize = costing_->UnitSize();
   float range = kBucketCount * bucketsize;
   adjacencylist_.reuse(mincost, range, bucketsize, &edgelabels_);
+  // adjacencylist_rev_.reuse(mincost, range, bucketsize, &edgelabels_rev_);
```

Comment:
can this be removed or are we keeping it around for reference?

Reply:
Removed, it's stale.